2021-APR /answer.md

```diff
@@ -0,0 +1,168 @@
+<!-- @import "[TOC]" {cmd="toc" depthFrom=1 depthTo=6 orderedList=false} -->
+
+# AACS1084 October 2021 Answers
+
+[Link to the paper](https://eprints.tarc.edu.my/17763/1/AACS1084.pdf)
+
+- [Question 1](#question-1)
+- [Question 2](#question-2)
+- [Question 3](#question-3)
+- [Question 4](#question-4)
+
+## Answers
+
+### Question 1
+
+a) (i)
+
+```c
+struct Singer {
+  char name[30];
+  char gender;
+  char nationality[30];
+};
+```
+
+b)
+
+```c
+typedef struct {
+  int id;
+  char title[30];
+  struct Singer singer;
+  double price;
+} Album;
+```
+
+c)
+
+```c
+Album cd[2] = {
+  {1001, "Happy Moments", "Jane Berry", 'F', "American", 67.90},
+  {1002, "Dreams", "Joseph Chan", 'M', "Taiwanese", 59.90}
+};
+```
+
+d)
+
+```c
+strcpy(cd[1].singer.nationality, "Singaporean");
+```
+
+e)
+
+```c
+for (int i = 0; i < 2; i++) {
+  printf("%d\n", cd[i].id);
+  printf("%s\n", cd[i].title);
+  printf("%s\n", cd[i].singer.name);
+  printf("%f\n", cd[i].price);
+}
+system("pause");
+```
+
+### Question 2
+
+a)
+
+```c
+FILE *fptr1, *fptr2;
+fptr1 = fopen("inventory.txt", "r");
+fptr2 = fopen("replenishment.txt", "w");
+```
+
+b)
+
+```c
+if (fptr1 == NULL) {
+  if (fptr2 == NULL) {
+    printf("Error : unable to open the file!");
+    exit(-1);
+  }
+}
+```
+
+c)
+
+```c
+//char prodID[5];
+//char prodName[30];
+//int prodQty, minStockLvl;
+
+while (fscanf(fptr1, "%[\|]|%[\|]|%[\|]|%[\n]\n", prodID, prodName, &prodQty, &minStockLvl) != EOF) {
+  if (prodQty < minStockLvl)
+    fprintf(fptr2, "%s|%s|%d|%d\n", prodID, prodName, prodQty, minStockLvl);
+}
+```
+
+d)
+
+```c
+fclose(fptr1);
+fclose(fptr2);
+```
+
+### Question 3
+
+a)
+```c
+FILE *fp;
+
+fp = fopen("employeeInfo.bin", "ab");
+fwrite(&emp, sizeof(Employee), 1, fp);
+
+fp = fopen("employeeInfo.bin", "rb");
```

Comment:
`fclose()` first before reopen the file, so the appended result is saved/written into the disk